src/commands/dependencies/__tests__/upload.test.ts

```diff
@@ -0,0 +1,405 @@
+    const filePath = './src/commands/dependencies/__tests__/fixtures/unknown-dependencies'
+    const resolvedFilePath = path.resolve(filePath)
+    const {context, code} = await runUploadDependenciesCommand(filePath, {
+      apiKey: 'DD_API_KEY_EXAMPLE',
+      appKey: 'DD_APP_KEY_EXAMPLE',
+      dryRun: true,
+      releaseVersion: '1.234',
+      service: 'my-service',
+      source: 'snyk',
+    })
+    const stdout = context.getStdoutBuffer().split(os.EOL)
+    const stderr = context.getStderrBuffer().split(os.EOL)
+
+    expect(stderr[0]).toEqual(chalk.red(`Cannot find "${resolvedFilePath}" file.`))
+    expect(stdout).toEqual([''])
+
+    expect(code).toBe(2)
+  })
+
+  test('shows warning if missing --release-version parameter', async () => {
+    const {context, code} = await runUploadDependenciesCommand(
+      './src/commands/dependencies/__tests__/fixtures/dependencies',
+      {
+        apiKey: 'DD_API_KEY_EXAMPLE',
+        appKey: 'DD_APP_KEY_EXAMPLE',
+        dryRun: true,
+        service: 'my-service',
+        source: 'snyk',
+      }
+    )
+    const stdout = context.getStdoutBuffer().split(os.EOL)
+    const stderr = context.getStderrBuffer().split(os.EOL)
+
+    expect(stderr).toEqual([''])
+    const releaseVersion = chalk.bold('--release-version')
+    expect(stdout[0]).toEqual(
+      chalk.yellow('┌──────────────────────────────────────────────────────────────────────────────────────┐')
+    )
+    expect(stdout[1]).toEqual(
+      chalk.yellow(`│ Missing optional ${releaseVersion} parameter.                                        │`)
+    )
+    expect(stdout[2]).toEqual(
+      chalk.yellow('│ The analysis may use out of date dependencies and produce false positives/negatives. │')
+    )
+    expect(stdout[3]).toEqual(
+      chalk.yellow('└──────────────────────────────────────────────────────────────────────────────────────┘')
+    )
+
+    expect(code).toBe(0)
+  })
+
+  test('makes a valid API request', async () => {
+    const filePath = './src/commands/dependencies/__tests__/fixtures/dependencies'
+    const resolvedFilePath = path.resolve(filePath)
+
+    const {context, code} = await runUploadDependenciesCommand(filePath, {
+      apiKey: 'DD_API_KEY_EXAMPLE',
+      appKey: 'DD_APP_KEY_EXAMPLE',
+      releaseVersion: '1.234',
+      service: 'my-service',
+      source: 'snyk',
+    })
+
+    const stdout = context.getStdoutBuffer().split(os.EOL)
+    const stderr = context.getStderrBuffer().split(os.EOL)
+
+    expect(stderr).toEqual([''])
+    expect(stdout[0]).toEqual(`${chalk.bold('File')}:    ${resolvedFilePath}`)
+    expect(stdout[1]).toEqual(`${chalk.bold('Source')}:  snyk`)
+    expect(stdout[2]).toEqual(`${chalk.bold('Service')}: my-service`)
+    expect(stdout[3]).toEqual(`${chalk.bold('Version')}: 1.234`)
+    expect(stdout[4]).toEqual('')
+    expect(stdout[5]).toEqual('Uploading dependencies...')
+    expect(stdout[6]).toMatch(/Dependencies uploaded in .* seconds\./)
```

Comment:
Same as before, this feels like a huge overhead to test every line of the output.
Especially including chalk's characters.

Reply:
I would like to test if everything is printed correctly to the end-user. I will make it looser by not checking line number and stripping chalk formatting :)